stdlib/domain.mli

```diff
@@ -91,6 +91,18 @@ val recommended_domain_count : unit -> int
 
     The value returned is at least [1]. *)
 
+val self_index : unit -> int
```

Comment:
Nitpick about naming and uniformity. Why is this called `self_index` and not just `index`, when the C function is named `caml_domain_index` and `caml_ml_domain_index`?

Reply:
This is more consistent with the Domain API which exposes `Domain.self : unit -> id` and `Domain.get_id : Domain.t -> id`. In the future we might want to introduce a `Domain.index` or `Domain.get_index` function with type `Domain.t -> int`, and I think that `self_index` is less likely to be confusing.